### PR TITLE
Fix bug in execution of TTL GROUP BY

### PR DESCRIPTION
**Pull request by alesapin**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Bug Fix

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix bug in `TTL` with `GROUP BY` expression which refuses to execute `TTL` after first execution in part.

No backport because of changes in `ttl.txt`. Finally fixes `test_replicated_ttl/test.py::test_ttl_compatibility`.

### CODE REVIEW

**Comment by alesapin** [comment]
No related failures.

**Comment by alesapin** [comment]
@mergifyio update
one more time

**Comment by alesapin** [comment]
Looks like in a very rare cases 01280_ttl_where_group_by completely ignores TTL

**Comment by alesapin** [comment]
@Mergifyio update

File: src/Storages/MergeTree/TTLMergeSelector.cpp [hunk 1]
```diff
@@ -111,6 +111,9 @@ bool TTLDeleteMergeSelector::isTTLAlreadySatisfied(const IMergeSelector::Part &
     if (only_drop_parts)
         return false;
 
+    if (!part.ttl_infos->hasAnyNonFinishedTTLs())
+        return false;
```
Comment (CurtizJ) [inline src/Storages/MergeTree/TTLMergeSelector.cpp:115]: Why we return `false` in this case?

**Review by CurtizJ — COMMENTED** [review]
You set `finished` flag if `isMinTTLExpired` is true. Looks like it can lead to non-pruned partially expired parts. Shouldn't we use `isMaxTTLExpired` instead?

**Review by CurtizJ — APPROVED** [review]

**Comment by alesapin** [comment]
No related failures.

**Pull request merged**